app/src/main/java/com/example/newsportal/app/topnews/CategoryFragment.kt

```diff
@@ -6,39 +6,41 @@ import android.view.View
 import android.view.ViewGroup
 import androidx.core.os.bundleOf
 import androidx.fragment.app.Fragment
-import androidx.recyclerview.widget.RecyclerView
 import com.example.newsportal.utils.NewsAdapter
-import com.example.newsportal.R
 import com.example.newsportal.data.local.model.ArticleLocal
+import com.example.newsportal.databinding.FragmentCategoryBinding
+import com.example.newsportal.domain.model.Article
 import org.koin.androidx.viewmodel.ext.android.getViewModel
 
 class CategoryFragment : Fragment() {
 
     private val viewModel: NewsViewModel by lazy { requireParentFragment().getViewModel<NewsViewModel>() }
     private val category: String by lazy { requireArguments().getString(CATEGORY_KEY, "general") }
+    private var _binding: FragmentCategoryBinding? = null
+    private val binding get() = _binding!!
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        return inflater.inflate(R.layout.fragment_category, container, false)
+        _binding = FragmentCategoryBinding.inflate(inflater, container, false)
+        return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        val recyclerView: RecyclerView = view.findViewById(R.id.rv)
         val adapter = NewsAdapter()
-        recyclerView.adapter = adapter
+        binding.rv.adapter = adapter

```

Comment:
Use more meaningful names for resource ids, "Component""Usecase". e.g. rvNews, tvTitle and so on